### ISSUE TITLE
Disjunctions in `if` conditions cause unbranching errors

### ISSUE THREAD

**Issue by jimbxb** [pschachte/wybe#218]
Example:
 ```
?a = true; ?b = false
if {~a | b :: pass }
```
Error:
```
Internal error: SemiDet proc call {wybe.bool.<0>~(a:wybe.bool @testing:29:6)} in a Det context
CallStack (from HasCallStack):
  error, called at src/AST.hs:3428:17 in main:AST
```

This is not the case with conjunctions, as this compiles fine:
```
?a = true; ?b = false
if {~a & b :: pass }
```

**Comment by jimbxb** [comment]
Interestingly, this case also compiles fine:
```
if {~true | false :: pass }
```

**Comment by pschachte** [comment]
Closed by PR #247
